src/meson.build

```diff
@@ -16,13 +16,23 @@ liblxi_deps = [
 tirpc_excluded_systems = ['darwin']
 liblxi_link_args = ['-fvisibility=hidden']
 
-# without avahi or bonjour, mdns discovery won't do anything
-if meson.get_compiler('c').has_header('avahi-client/client.h')
-  add_project_arguments('-DHAVE_AVAHI', language: 'c')
-  liblxi_sources += 'avahi.c'
-elif meson.get_compiler('c').has_header('dns_sd.h')
-  add_project_arguments('-DHAVE_BONJOUR', language: 'c')
-  liblxi_sources += 'bonjour.c'
+if(build_machine.system() == 'cygwin')
+  liblxi_link_args = ['-lws2_32']
+
+  inc_dir = include_directories('.')
```

Comment:
I don't think you need this line.

Reply:
You don't really need to do the check for cygwin/dns_sh.h as it is now shipped with the source.